### PR TITLE
docs(website): add DocSearch as recommended by docusaurus.

### PR DESCRIPTION
**Pull request by s-pace**
👋 team,

I am working on DocSearch. [We have an integration for Docusaurus](https://docusaurus.io/docs/en/search#enabling-the-search-bar). We thought it is a shame you can not also used this search that you can [find on reactjs for example](https://reactjs.org/).

This PR will add DocSearch to the documentation website. It will allow an user to have a learn-as-you-type experience by displaying results thanks to a dropdown in a live way.

Let me know if you need anything.

### CODE REVIEW

**Comment by jrwats** [comment]
LGTM, algolia, eh?

**Comment by jrwats** [comment]
@s-pace 
whoops, I trusted that this was going to work, but:

```
>yarn build
yarn run v1.21.1
$ docusaurus build
Creating an optimized production build...
Error: The field(s) 'algolia' are not recognized in docusaurus.config.js
    at Object.loadConfig (/Users/jwatson/workspace/fbt/website/node_modules/@docusaurus/core/lib/server/config.js:55:15)
    at loadContext (/Users/jwatson/workspace/fbt/website/node_modules/@docusaurus/core/lib/server/index.js:25:33)
    at Object.load (/Users/jwatson/workspace/fbt/website/node_modules/@docusaurus/core/lib/server/index.js:51:21)
    at build (/Users/jwatson/workspace/fbt/website/node_modules/@docusaurus/core/lib/commands/build.js:52:34)
    at args (/Users/jwatson/workspace/fbt/website/node_modules/@docusaurus/core/bin/docusaurus.js:29:5)
    at Command.cli.command.description.option.action (/Users/jwatson/workspace/fbt/website/node_modules/@docusaurus/core/bin/docusaurus.js:45:23)
    at Command.listener (/Users/jwatson/workspace/fbt/website/node_modules/commander/index.js:360:8)
    at Command.emit (events.js:182:13)
    at Command.parseArgs (/Users/jwatson/workspace/fbt/website/node_modules/commander/index.js:799:12)
    at Command.parse (/Users/jwatson/workspace/fbt/website/node_modules/commander/index.js:563:21)
    at Object.<anonymous> (/Users/jwatson/workspace/fbt/website/node_modules/@docusaurus/core/bin/docusaurus.js:97:5)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
    at Module.load (internal/modules/cjs/loader.js:612:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:551:12)
    at Function.Module._load (internal/modules/cjs/loader.js:543:3)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by s-pace** [comment]
Weird @jrwats , it is documented here 
https://docusaurus.io/docs/en/site-config#algolia-object

**Comment by jrwats** [comment]
@ozgunbal, @yangshun, @endiliey, 
Is this, perhaps related to differences in Docusaurus v1 => v2?

**Comment by yangshun** [comment]
Not really. We moved the location of the value into `themeConfig` now because it's a theme thing.

File: website/docusaurus.config.js [hunk 1]
```diff
@@ -95,6 +95,10 @@ module.exports = {
     },
     image: "img/fbt.png"
   },
+  algolia: {
```
Comment (yangshun) [inline website/docusaurus.config.js]: Should be under `themeConfig`.

**Pull request closed without merging**